CommonTools/CandAlgos/interface/GenParticleCustomSelector.h

```diff
@@ -42,19 +65,38 @@ class GenParticleCustomSelector {
           testId = true;
       }
 
-    return (tp.pt() >= ptMin_ && tp.eta() >= minRapidity_ && tp.eta() <= maxRapidity_ &&
-            sqrt(tp.vertex().perp2()) <= tip_ && fabs(tp.vertex().z()) <= lip_ && tp.status() == status_ && testId);
+    auto etaOk = [&](const reco::GenParticle& p) -> bool {
+      float eta = p.eta();
+      if (!invertRapidityCut_)
+        return (eta >= minRapidity_) & (eta <= maxRapidity_);

```

Comment:
Please use logical operators ("&&", "||") instead of bitwise operators ("&", "|"), here and everywhere else in your updates